types/utils/event.d.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare global {
+  export interface WMouseEvent<T extends HTMLElement> extends MouseEvent {
```

Comment:
Initially I thought this was referring to a mouse **W**heel event.

Maybe the type could hint at why it's needed. Like `MouseEventWithTarget` or something like that?

Reply:
I was confused too but I actually really like this type. It seems like an obvious addition to `Event`.

Reply:
oh yeah I'm cool with adding the type, but I'm wondering if its name can be more descriptive. At a glance it's not clear what WMouseEvent means.

Reply:
I went with `W` as we use it as a prefix for our classes. I can rename it if there's a preference.